src/memusage.h

```diff
@@ -106,6 +115,21 @@ static inline size_t IncrementalDynamicUsage(const std::map<X, Y, Z>& m)
     return MallocUsage(sizeof(stl_tree_node<std::pair<const X, Y> >));
 }
 
+template<typename X>
+static inline size_t DynamicUsage(const std::unique_ptr<X>& p)
+{
+    return p ? MallocUsage(sizeof(X)) : 0;
+}
+
+template<typename X>
+static inline size_t DynamicUsage(const std::shared_ptr<X>& p)
+{
+    // A shared_ptr can either use a single continuous memory block for both
+    // the counter and the storage (when using std::make_shared), or separate.
+    // We can't observe the difference, however, so assume the worst.
+    return p ? MallocUsage(sizeof(X)) + MallocUsage(sizeof(stl_shared_counter<int>)) : 0;
```

Comment:
Is there any reference for these assumptions?  `stl_shared_counter` looks "right",  but short on investigating the inner workings of my local STL,  is this guaranteed?